pkg/sysched/sysched.go

```diff
@@ -0,0 +1,505 @@
+package sysched
+
+//TODO:
+// 1. weighted mechanism: i) extending SPO, ii) overload existing SPO, iii) CRD
+// 2. handling the restart of scheduler to restore states
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"path"
+	"strings"
+
+	"github.com/containers/common/pkg/seccomp"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/helper"
+	pluginconfig "sigs.k8s.io/scheduler-plugins/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/sysched/clientset/v1alpha1"
+	"sigs.k8s.io/security-profiles-operator/api/seccompprofile/v1beta1"
+)
+
+type SySched struct {
+	handle                  framework.Handle
+	clientSet               v1alpha1.SPOV1Alpha1Interface
+	HostToPods              map[string][]*v1.Pod
```

Comment:
As explained above, we keep a local map of HostToPods in our plugin rather than rely on the cached state to reliably keep track of all pods on the cluster, regardless of which scheduler it used.